### PR TITLE
Support token usage in streaming

### PR DESCRIPTION
**Pull request by B-Step62**
Resolve https://github.com/langchain-ai/langchain-databricks/issues/9

[ChatOpenAI](https://github.com/langchain-ai/langchain/blob/e62e390ca01dff3e9e20e9fcc9a12520b5ec7931/libs/partners/openai/langchain_openai/chat_models/base.py#L1854-L1876) support streaming token usage, `ChatDatabricks` should be compatible with it.

**Test**
- Unit test
- Integration test against FMAPI endpoint.
- Manual test

Screenshot 2024-10-21 at 14 32 14

### CODE REVIEW

File: libs/databricks/langchain_databricks/chat_models.py [hunk 1]
```diff
@@ -313,8 +346,20 @@ def _stream(
                 if first_chunk_role is None:
                     first_chunk_role = chunk_delta.get("role")
 
+                if stream_usage:
```
Comment (harupy) [inline libs/databricks/langchain_databricks/chat_models.py:349]: do we need this flag? can we always include the usage data?

Comment (B-Step62) [reply libs/databricks/langchain_databricks/chat_models.py:349]: Usage data is optional for streaming in all chat models.

Comment (harupy) [reply libs/databricks/langchain_databricks/chat_models.py:349]: Got it, just wanted to make sure this aligns with other chat models.

Comment (harupy) [inline libs/databricks/langchain_databricks/chat_models.py:349]:
```suggestion
                if stream_usage and (usage := chunk.get("usage")):
```


can we only include `usage` when it's present?

File: libs/databricks/langchain_databricks/chat_models.py [hunk 2]
```diff
@@ -759,7 +804,9 @@ def _convert_dict_to_message(_dict: Dict) -> BaseMessage:
 
 
 def _convert_dict_to_message_chunk(
-    _dict: Mapping[str, Any], default_role: str
+    _dict: Mapping[str, Any],
+    default_role: str,
+    usage: Optional[Dict] = None,
```
Comment (harupy) [inline libs/databricks/langchain_databricks/chat_models.py:809]:
```suggestion
    usage: Optional[Dict[str, Any]] = None,
```

nit

**Review by harupy — APPROVED** [review]
LGTM once comments are addressed.

File: libs/databricks/langchain_databricks/chat_models.py (hunk 1 above)
Comment (B-Step62) [reply libs/databricks/langchain_databricks/chat_models.py:349]: makes sense!

**Pull request merged**